File: wsTripXML/Code/OTAClasses/OTAMultiMessageRQ.vb

Option Strict Off
Option Explicit On 

Imports System.Xml.Serialization

Namespace wsTravelTalk.wmMultiMessageIn

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)> _
    Public Class MultiMessageRQ

        '<remarks/>
        Public POS As POSType

        '<remarks/>
        Public MultiMessage As String
    End Class

    '<remarks/>
    Public Class POSType

        '<remarks/>
        <System.Xml.Serialization.XmlElementAttribute("Source")> _
        Public Source() As SourceType

        '<remarks/>
        Public TPA_Extensions As TPA_ExtensionsType
    End Class

    '<remarks/>
    Public Class SourceType

        '<remarks/>
        Public RequestorID As RequestorIDType

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public PseudoCityCode As String
    End Class

    '<remarks/>
    Public Class RequestorIDType

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public Type As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()> _
        Public ID As String
    End Class

    '<remarks/>
    Public Class ProviderType

        '<remarks/>
        Public Name As String

        '<remarks/>
        Public System As String

        '<remarks/>
        Public Userid As String

        '<remarks/>
        Public Password As String
    End Class

    '<remarks/>
    Public Class TPA_ExtensionsType

        '<remarks/>
        Public Provider As ProviderType
    End Class
End Namespace

File: src/wsTripXML/Code/OTAClasses/OTAMultiMessageRQ.cs
using System.Xml.Serialization;

namespace wsTripXML.wsTravelTalk.wmMultiMessageIn
{

    [XmlRoot(IsNullable = false)]
    public class MultiMessageRQ
    {

        public POSType POS;

        public string MultiMessage;
    }

    public class POSType : Code.IPOS
    {

        public TPA_Extensions TPA_Extensions;
    }

    public class SourceType
    {

        public RequestorIDType RequestorID;

        [XmlAttribute()]
        public string PseudoCityCode;
    }

    public class RequestorIDType
    {

        [XmlAttribute()]
        public string Type;

        [XmlAttribute()]
        public string ID;
    }

    public class ProviderType
    {

        public string Name;

        public string System;

        public string Userid;

        public string Password;
    }

    public class TPA_Extensions : Code.ITPA_Extensions
    {

        public ProviderType Provider;
    }
}